app-backend/datamodel/src/main/scala/Scope.scala

```diff
@@ -0,0 +1,308 @@
+package com.rasterfoundry.datamodel
+
+import cats.{Eq, Monoid}
+import cats.implicits._
+import io.circe.{Decoder, DecodingFailure, Encoder, HCursor, Json}
+import io.circe.parser._
+
+import scala.util.{Failure, Success}
+
+case class Action(domain: String, action: String, limit: Option[Long]) {
```

Comment:
Downside of including the limit here is that we won't be able to use the fast `contains` that `Set`s have (I think -- unless providing an `Eq` instance for `Action` influences `Set` behavior, which I don't think it does). I think probably it will be the case that we won't have so many actions that we'll end up with a big penalty, but here are some example timings of the potential risk:

```scala
@ time(smallSet(3)) 
res4: (Boolean, concurrent.duration.FiniteDuration) = (true, 29135 nanoseconds)
@ time(biggerSet(3)) 
res5: (Boolean, concurrent.duration.FiniteDuration) = (true, 34508 nanoseconds)
@ time(biggestSet(3)) 
res6: (Boolean, concurrent.duration.FiniteDuration) = (true, 29974 nanoseconds)
@ time(smallSet.filter(_ == 3)) 
res8: (Set[Int], concurrent.duration.FiniteDuration) = (Set(3), 297819 nanoseconds)
@ time(biggerSet.filter(_ == 3)) 
res9: (Set[Int], concurrent.duration.FiniteDuration) = (Set(3), 2360021 nanoseconds)
@ time(biggestSet.filter(_ == 3)) 
res10: (Set[Int], concurrent.duration.FiniteDuration) = (Set(3), 129637066 nanoseconds)
```

where `smallSet` is a `Set` containing ints 1 to 10, `biggerSet` is a set containing ints 1 to 1000, and `biggestSet` is a set containing ints 1 to 1 million. No one should end up with 1 million actions (or even 100k actions), so it should be safe to filter the set on `domain` and `action` and then get the quantitative limit out.